solidity/contracts/bridge/WalletProposalValidator.sol

```diff
@@ -587,6 +597,71 @@ contract WalletProposalValidator {
         return true;
     }
 
+    /// @notice View function encapsulating the main rules of a valid moving
+    ///         funds proposal. This function is meant to facilitate the
+    ///         off-chain validation of the incoming proposals. Thanks to it,
+    ///         most of the work can be done using a single readonly contract
+    ///         call.
+    /// @param proposal The moving funds proposal to validate.
+    /// @return True if the proposal is valid. Reverts otherwise.
+    /// @dev Notice that this function is meant to be invoked after the moving
+    ///      funds commitment has already been submitted. This function skips
+    ///      some checks related to the moving funds procedure as they were
+    ///      already checked on the commitment submission.
+    ///      Requirements:
+    ///      - The source wallet must be in the MovingFunds state,
+    ///      - The target wallets commitment must be submitted,
+    ///      - The target wallets commitment hash must match the target wallets
+    ///        from the proposal,
+    ///      - The proposed moving funds transaction fee must be greater than
+    ///        zero,
+    ///      - The proposed moving funds transaction fee must not exceed the
+    ///        maximum total fee allowed for moving funds.
+    function validateMovingFundsProposal(MovingFundsProposal calldata proposal)
```

Comment:
I think there is an additional thing that should be checked in this function. Namely:
```
require(
    walletBtcBalance >= movingFundsDustThreshold,
    "Wallet BTC balance must be equal or above the moving funds dust threshold"
);
```

This is because there is the `notifyMovingFundsBelowDust` for wallets whose balance is below the dust. If we don't check that here, the following may happen:
1. The leader submits a commitment
2. The leader creates a proposal
3. The leader validates the proposal using `validateMovingFundsProposal`
4. The leader broadcasts the proposal
5. Followers receive the proposal and validate it using `validateMovingFundsProposal`
6. Followers start the action that leads to a BTC moving funds transaction
7. A third party calls `notifyMovingFundsBelowDust` that moves the wallet to the `Closing` state
8. Submitting the SPV through `submitMovingFundsProof` is no longer possible as the wallet must be in the `MovingFunds` state
9. A third party accuses the wallet of committing fraud. The wallet cannot defeat the challenge as the SPV proof cannot be submitted so the main UTXO was spent illegally

The leader will not broadcast the proposal if we add the aforementioned `require` to `validateMovingFundsProposal`. It can still submit the commitment (because that happens before proposal validation, see https://github.com/keep-network/tbtc-v2/pull/757#discussion_r1431320949) but that's not a problem. On the other hand, if a leader broadcasts the proposal regardless, the followers will reject it due to this `require`.

Please verify my understanding. We can merge this PR to not block it and add this check in a follow-up if you find the above scenario correct.

Reply:
Done in db47dbb09b123424f39a5d6c3f2c75a635c04d57.
This check indeed seems to be necessary.